### ISSUE TITLE
🐛 Bug: migration file "nc_050_tenant_isolation" failed (`value too long for type character varying(20)`)

### ISSUE THREAD

**Issue by nil0x42** [nocodb/nocodb#8773]
### Please confirm if bug report does NOT exist already ?

- [X] I confirm there is no existing issue for this

### Steps to reproduce ?

Migrate to v0.250.2 docker image, start nocodb.
Logs:
```
nc_050_tenant_isolation: Migration started
nc_050_tenant_isolation: Adding missing base_id columns
nc_050_tenant_isolation: Added missing base_id columns in 0.031844237s
nc_050_tenant_isolation: Migrating data
nc_050_tenant_isolation: Migrated base_id for nc_calendar_view_range_v2 in 0.001162959s
nc_050_tenant_isolation: Migrated base_id for nc_col_barcode_v2 in 0.00068004s
nc_050_tenant_isolation: Migrated base_id for nc_col_formula_v2 in 0.000540137s
nc_050_tenant_isolation: Migrated base_id for nc_col_lookup_v2 in 0.000594898s
nc_050_tenant_isolation: Migrated base_id for nc_col_qrcode_v2 in 0.000491004s
nc_050_tenant_isolation: Migrated base_id for nc_col_relations_v2 in 0.000956786s
nc_050_tenant_isolation: Migrated base_id for nc_col_rollup_v2 in 0.000551733s
migration file "nc_050_tenant_isolation" failed
migration failed with error:
        UPDATE nc_col_select_options_v2
        SET base_id = nc_columns_v2.base_id
        FROM nc_columns_v2
        WHERE nc_col_select_options_v2.fk_column_id = nc_columns_v2.id
       - value too long for type character varying(20)
```

### Desired Behavior

Migration succeeds

### Project Details

used as docker: True (latest docker image 0.250.2)
NC_DB: pg
Project created connecting to external database.
Database on which spreadsheet is created: pg

**Comment by candideu** [comment]
I'm having the same issues... Trying to upgrade from 0.207.3 to 0.250.2 (had successfully upgraded from 205.1 beforehand).

```sql
[Nest] 9  - 06/17/2024, 10:26:11 PM   ERROR [ExceptionHandler] 
UPDATE nc_col_relations_v2
SET base_id = nc_columns_v2.base_id
FROM nc_columns_v2
WHERE nc_col_relations_v2.fk_column_id = nc_columns_v2.id
- value too long for type character varying(20)
error:
UPDATE nc_col_relations_v2
SET base_id = nc_columns_v2.base_id
FROM nc_columns_v2
WHERE nc_col_relations_v2.fk_column_id = nc_columns_v2.id
- value too long for type character varying(20)
at Parser.parseErrorMessage (/usr/src/app/node_modules/pg-protocol/dist/parser.js:283:98)
at Parser.handlePacket (/usr/src/app/node_modules/pg-protocol/dist/parser.js:122:29)
at Parser.parse (/usr/src/app/node_modules/pg-protocol/dist/parser.js:35:38)
at Socket.<anonymous> (/usr/src/app/node_modules/pg-protocol/dist/index.js:11:42)
at Socket.emit (node:events:518:28)
at Socket.emit (node:domain:488:12)
at addChunk (node:internal/streams/readable:559:12)
at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
at Readable.push (node:internal/streams/readable:390:5)
at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
```

**Comment by mertmit** [comment]
Hi thanks for reporting this issue, I will need some more information.

Can you please share result of following query in your NC_DB:
```
SELECT id FROM nc_bases_v2 WHERE LENGTH(id) > 20;
```

**Comment by o1lab** [comment]
We suspect the issue might be due to ID being generated on client side (likely from your API usage) instead of server side generated and they are beyond 20 characters. We advise you to not send your own IDs while creating any of the models.

**Comment by nil0x42** [comment]
@mertmit , here is the output. Indeed there is a base with such long name:
```
nocodb@localhost:nocodb> SELECT id FROM nc_bases_v2 WHERE LENGTH(id) > 20;
+--------------------------------------+
| id                                   |
|--------------------------------------|
| ribot_data_analysis_information__22n |
+--------------------------------------+
SELECT 1
Time: 0.010s
```

**Comment by candideu** [comment]
My error is now solved by upgrading to 0.251.1. Thanks team!
